File: Python/ЕГЭ2021;2022/27/Решение заданий. Поляков/4514.py
f = open('27-81bb.txt')
n = int(f.readline())

z = [float('inf')]*13
k = s = answer = 0

for _ in range(n):
    x = int(f.readline())
    s+=x
    
    if x % 2 != 0: k+=1
    if k % 13 == 0:
        answer = s
    elif (z[k % 13] != float('inf')):
        answer = max(answer, s-z[k % 13])

    z[k % 13] = min(z[k % 13], s)
    
print(answer)

File: C++/ЕГЭ2021;2022/27/Решение заданий. Поляков/4514.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
	int n;
	int s = 0;
	int k = 0;
	int answer = 0;
	vector<int> arr(13, 100000);

	fstream f("C:\\Users\\admin\\Downloads\\27-81bb.txt");
	f >> n;

	for (int i = 0; i < n; i++)
	{
		int x;
		f >> x;
		s += x;
		if (x % 2 != 0)
			k++;
		if (k % 13 == 0)
		{
			answer = s;

		}
		else if (arr[k % 13] != 0) {
			answer = max(answer, s - arr[k % 13]);
		}

		arr[k % 13] = min(arr[k % 13], s);
	}

	cout << answer;


	return 0;
}